ionoscloud/resource_nfs_share.go

```diff
@@ -0,0 +1,217 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services"
+)
+
+func resourceNFSShare() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNFSShareCreate,
+		ReadContext:   resourceNFSShareRead,
+		UpdateContext: resourceNFSShareUpdate,
+		DeleteContext: resourceNFSShareDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceNFSShareImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Description: "The ID of the Network File Storage Share.",
+				Computed:    true,
+			},
+			"cluster_id": {
+				Type:        schema.TypeString,
+				Description: "The ID of the Network File Storage Cluster.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"location": {
+				Type: schema.TypeString,
+				Description: fmt.Sprintf("The location of the Network File Storage Cluster. "+
+					"Available locations: '%s'", strings.Join(ValidNFSLocations, ", '")),
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(ValidNFSLocations, false)),
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The directory being exported",
+				Required:    true,
+			},
+			"nfs_path": {
+				Type:        schema.TypeString,
+				Description: "Path to the NFS export. The NFS path is the path to the directory being exported.",
+				Computed:    true,
+			},
+			"quota": {
+				Type:        schema.TypeInt,
+				Description: "The quota in MiB for the export. The quota can restrict the amount of data that can be stored within the export. The quota can be disabled using `0`.",
+				Optional:    true,
+				Default:     0,
+			},
+			"gid": {
+				Type:        schema.TypeInt,
+				Description: "The group ID that will own the exported directory. If not set, **anonymous** (`512`) will be used.",
+				Optional:    true,
+				// Default:     512, // max 65534
```

Comment:
Any specific reason for `Default` being commented out?

Reply:
I know this default in particular is unlikely to change, but afaik the common consensus was to leave it up to the API to handle optionals, and simply send them as null.